src/aotcompile.cpp

```diff
@@ -404,16 +404,18 @@ void *jl_create_native(jl_array_t *methods, const jl_cgparams_t cgparams, int _p
 
     // move everything inside, now that we've merged everything
     // (before adding the exported headers)
-    for (GlobalObject &G : clone->global_objects()) {
-        if (!G.isDeclaration()) {
-            G.setLinkage(Function::InternalLinkage);
-            makeSafeName(G);
-            addComdat(&G);
+    if (policy != CompilationPolicy::Extern) {
```

Comment:
```suggestion
    if (policy == CompilationPolicy::Default) {
```

So that this doesn't conflict with https://github.com/JuliaLang/julia/pull/38642